lib/resourcemerge/security.go

```diff
@@ -44,7 +45,14 @@ func EnsureSecurityContextConstraints(existing securityv1.SecurityContextConstra
 	setBool(&modified, &result.AllowHostPID, required.AllowHostPID)
 	setBool(&modified, &result.AllowHostIPC, required.AllowHostIPC)
 	setBoolPtr(&modified, &result.DefaultAllowPrivilegeEscalation, required.DefaultAllowPrivilegeEscalation)
-	setBoolPtr(&modified, &result.AllowPrivilegeEscalation, required.AllowPrivilegeEscalation)
+
+	// AllowPrivilegeEscalation is optional and defaults to true if not specified,
+	// so we enforce default if manifest does not specify it.
+	if required.AllowPrivilegeEscalation == nil {
+		setBoolPtr(&modified, &result.AllowPrivilegeEscalation, pointer.Bool(true))
+	} else {
+		setBoolPtr(&modified, &result.AllowPrivilegeEscalation, required.AllowPrivilegeEscalation)
+	}
```

Comment:
I do not see other ways to fix the issue. Hopefully this doe not break anything.